subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -17,37 +17,145 @@
 #
 
 ##############################################################################
-##
-##  ${applicationName} start up script for UN*X
-##
+#
+#   ${applicationName} start up script for POSIX<% if ( applicationName != 'Gradle' ) { %>
+#   generated by Gradle<% } %>.
+#
+#   You're probably reading this because you're a Unix/Linux person and have
+#   been instructed to run "gradlew" to install or build a project that's of
+#   interest to you. (At least, that's how I wound up reading this script.)
+#
+#   Here's some stuff you need to know:
+#
+#   (1) You need a POSIX-compliant shell to run this script. If your /bin/sh is
+#       noncompliant, but you have some other compliant shell such as ksh or
+#       bash, then to run this script, type that shell name before the whole
+#       command line, like:
+#
+#           ksh gradlew
+#
+#       Even if you have a compliant shell, Java is fussy about $PWD, so you
+#       might as well cd to the directory containing gradlew and run it as
+#
+#           ./gradlew
+#
+#       Any arguments you include on the command line will be passed through to
+#       Java after the classname, to be available as parameters to main().
+#
+#   (2) This script is generated from the Groovy template file
+#       subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt
+#       within the Gradle project, so that's where you need to submit any
+#       patches.
+#
+#       Gradle is a meta-build system used by the project that you're building
+#       or installing. It's like autoconf but for projects that are written in
+#       Java and related languages. It's also used to build parts of the Gradle
+#       project itself.
+#
+#       Gradle is mostly written in Java and Groovy, but there are wrapper
+#       scripts for various environments, including this one for POSIX shells.
```

Comment:
Since this is something that gets copied into a lot of projects, we need to be considerate with the amount of detail here.

I think the comments that are left in this file should describe how this script works and where this script comes from.

e.g., I was thinking of summarizing your (1)-(3) with:
```
#
#   (1) You need a POSIX-compliant shell to run this script. If your /bin/sh is
#       noncompliant, but you have some other compliant shell such as ksh or
#       bash, then to run this script, type that shell name before the whole
#       command line, like:
#
#           ksh $applicationName
# 
#   (2) This script targets any POSIX shell, not just Bash or Ksh.
#
#       It makes use of these POSIX shell features:
#         * functions;
#         * various expansions, including «$var», «$( cmd )», «${var#prefix}»,
#           and «${var%suffix}»;
#         * compound commands having a testable exit status, especially «case»;
#         * various built-in commands including «command», «set», and «ulimit».
#
#       However arrays are NOT available, except for "\$@".
#
#       This script progressively modifies "\$@", which is eventually
#       passed to Java.
#
#       There is special handling for the inherited environment variables
#       DEFAULT_JVM_OPTS, JAVA_OPTS, and ${optsEnvironmentVar}; see the in-line
#       comments for details.
#
#       There are tweaks for specific operating systems such as AIX, CygWin,
#       Darwin, MinGW, and NonStop.
#
#   (3) This script is generated from a Groovy template file in Gradle
#       https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt
#
```

Reply:
That looks like a reasonable balance.
Enough detail so that the reader of `gradlew` can understand it (and why it is written the way that it is), and a link so somewhere with more detail.